server/src/main/java/org/apache/druid/server/coordinator/compact/CompactionStatus.java

```diff
@@ -307,9 +327,16 @@ private CompactionStatus metricsSpecIsUpToDate()
       }
 
       final List<Object> metricSpecList = lastCompactionState.getMetricsSpec();
-      final AggregatorFactory[] existingMetricsSpec
-          = CollectionUtils.isNullOrEmpty(metricSpecList)
-            ? null : objectMapper.convertValue(metricSpecList, AggregatorFactory[].class);
+      final AggregatorFactory[] existingMetricsSpec;
+      if (CollectionUtils.isNullOrEmpty(metricSpecList)) {
```

Comment:
please add a comment here explaining what is being done.

Reply:
Added a comment.